File: dashboard/src/types/marketing-kb.ts
// Marketing Knowledge Base Types

export type MKBContentType = 'messaging' | 'case_study' | 'deck' | 'objection' | 'roi';
export type MKBPersona = 'director' | 'principal' | 'teacher' | 'parent' | 'it_admin' | 'county_official';
export type MKBContextTag = 'rural' | 'urban' | 'low_connectivity' | 'no_isp' | 'connected' | 'private' | 'public' | 'cbc' | 'igcse' | '8-4-4';
export type MKBArticleStatus = 'draft' | 'review' | 'approved' | 'archived';

export interface MKBArticle {
  id: string;
  tenantId: string;
  title: string;
  slug: string;
  summary: string;
  content: string;
  contentType: MKBContentType;
  personas: string[];
  contextTags: string[];
  tags: string[];
  version: number;
  status: MKBArticleStatus;
  usageCount: number;
  lastUsedAt?: string;
  createdById: string;
  createdByName: string;
  updatedById: string;
  updatedByName: string;
  approvedAt?: string;
  approvedById?: string;
  approvedByName?: string;
  createdAt: string;
  updatedAt: string;
}

export interface PitchKit {
  id: string;
  tenantId: string;
  name: string;
  description: string;
  targetPersona: string;
  contextTags: string[];
  articleIds: string[];
  articles?: MKBArticle[];
  isTemplate: boolean;
  createdById: string;
  createdByName: string;
  updatedById: string;
  updatedByName: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMKBArticleRequest {
  title: string;
  slug?: string;
  summary?: string;
  content: string;
  contentType?: MKBContentType;
  personas?: string[];
  contextTags?: string[];
  tags?: string[];
}

export interface UpdateMKBArticleRequest {
  title?: string;
  slug?: string;
  summary?: string;
  content?: string;
  contentType?: MKBContentType;
  personas?: string[];
  contextTags?: string[];
  tags?: string[];
  status?: 'draft' | 'review';
}

export interface CreatePitchKitRequest {
  name: string;
  description?: string;
  targetPersona?: string;
  contextTags?: string[];
  articleIds?: string[];
  isTemplate?: boolean;
}

export interface UpdatePitchKitRequest {
  name?: string;
  description?: string;
  targetPersona?: string;
  contextTags?: string[];
  articleIds?: string[];
  isTemplate?: boolean;
}

export interface MKBArticleFilters {
  q?: string;
  contentType?: MKBContentType;
  persona?: MKBPersona;
  contextTag?: MKBContextTag;
  status?: MKBArticleStatus;
  limit?: number;
  cursor?: string;
}

export interface PitchKitFilters {
  targetPersona?: MKBPersona;
  isTemplate?: boolean;
  limit?: number;
  cursor?: string;
}

export interface MKBStats {
  total: number;
  approved: number;
  inReview: number;
  draft: number;
  byContentType: Record<string, number>;
  byPersona: Record<string, number>;
  byContextTag: Record<string, number>;
}

// Display labels for enum values
export const mkbContentTypeLabels: Record<MKBContentType, string> = {
  messaging: 'Messaging',
  case_study: 'Case Study',
  deck: 'Deck',
  objection: 'Objection Handler',
  roi: 'ROI Calculator',
};

export const mkbContentTypeIcons: Record<MKBContentType, string> = {
  messaging: 'MessageSquare',
  case_study: 'BookOpen',
  deck: 'Presentation',
  objection: 'Shield',
  roi: 'Calculator',
};

export const mkbPersonaLabels: Record<MKBPersona, string> = {
  director: 'Director',
  principal: 'Principal',
  teacher: 'Teacher',
  parent: 'Parent',
  it_admin: 'IT Admin',
  county_official: 'County Official',
};

export const mkbContextTagLabels: Record<MKBContextTag, string> = {
  rural: 'Rural',
  urban: 'Urban',
  low_connectivity: 'Low Connectivity',
  no_isp: 'No ISP',
  connected: 'Connected',
  private: 'Private',
  public: 'Public',
  cbc: 'CBC',
  igcse: 'IGCSE',
  '8-4-4': '8-4-4',
};

export const mkbStatusLabels: Record<MKBArticleStatus, string> = {
  draft: 'Draft',
  review: 'In Review',
  approved: 'Approved',
  archived: 'Archived',
};

export const mkbStatusColors: Record<MKBArticleStatus, string> = {
  draft: 'bg-amber-100 text-amber-800',
  review: 'bg-blue-100 text-blue-800',
  approved: 'bg-green-100 text-green-800',
  archived: 'bg-gray-100 text-gray-800',
};

File: services/ims-api/internal/models/marketing_kb.go
package models

import "time"

// MKBContentType represents the type of marketing KB article
type MKBContentType string

const (
	MKBContentTypeMessaging MKBContentType = "messaging"
	MKBContentTypeCaseStudy MKBContentType = "case_study"
	MKBContentTypeDeck      MKBContentType = "deck"
	MKBContentTypeObjection MKBContentType = "objection"
	MKBContentTypeROI       MKBContentType = "roi"
)

// MKBPersona represents the target persona for the content
type MKBPersona string

const (
	MKBPersonaDirector       MKBPersona = "director"
	MKBPersonaPrincipal      MKBPersona = "principal"
	MKBPersonaTeacher        MKBPersona = "teacher"
	MKBPersonaParent         MKBPersona = "parent"
	MKBPersonaITAdmin        MKBPersona = "it_admin"
	MKBPersonaCountyOfficial MKBPersona = "county_official"
)

// MKBContextTag represents school context tags
type MKBContextTag string

const (
	MKBContextRural           MKBContextTag = "rural"
	MKBContextUrban           MKBContextTag = "urban"
	MKBContextLowConnectivity MKBContextTag = "low_connectivity"
	MKBContextNoISP           MKBContextTag = "no_isp"
	MKBContextConnected       MKBContextTag = "connected"
	MKBContextPrivate         MKBContextTag = "private"
	MKBContextPublic          MKBContextTag = "public"
	MKBContextCBC             MKBContextTag = "cbc"
	MKBContextIGCSE           MKBContextTag = "igcse"
	MKBContext844             MKBContextTag = "8-4-4"
)

// MKBArticleStatus represents the publication status
type MKBArticleStatus string

const (
	MKBStatusDraft    MKBArticleStatus = "draft"
	MKBStatusReview   MKBArticleStatus = "review"
	MKBStatusApproved MKBArticleStatus = "approved"
	MKBStatusArchived MKBArticleStatus = "archived"
)

// MKBArticle represents a marketing KB article
type MKBArticle struct {
	ID             string           `json:"id"`
	TenantID       string           `json:"tenantId"`
	Title          string           `json:"title"`
	Slug           string           `json:"slug"`
	Summary        string           `json:"summary"`
	Content        string           `json:"content"`
	ContentType    MKBContentType   `json:"contentType"`
	Personas       []string         `json:"personas"`
	ContextTags    []string         `json:"contextTags"`
	Tags           []string         `json:"tags"`
	Version        int              `json:"version"`
	Status         MKBArticleStatus `json:"status"`
	UsageCount     int              `json:"usageCount"`
	LastUsedAt     *time.Time       `json:"lastUsedAt,omitempty"`
	CreatedByID    string           `json:"createdById"`
	CreatedByName  string           `json:"createdByName"`
	UpdatedByID    string           `json:"updatedById"`
	UpdatedByName  string           `json:"updatedByName"`
	ApprovedAt     *time.Time       `json:"approvedAt,omitempty"`
	ApprovedByID   string           `json:"approvedById,omitempty"`
	ApprovedByName string           `json:"approvedByName,omitempty"`
	CreatedAt      time.Time        `json:"createdAt"`
	UpdatedAt      time.Time        `json:"updatedAt"`
}

// PitchKit represents a saved collection of marketing articles
type PitchKit struct {
	ID            string       `json:"id"`
	TenantID      string       `json:"tenantId"`
	Name          string       `json:"name"`
	Description   string       `json:"description"`
	TargetPersona string       `json:"targetPersona"`
	ContextTags   []string     `json:"contextTags"`
	ArticleIDs    []string     `json:"articleIds"`
	Articles      []MKBArticle `json:"articles,omitempty"`
	IsTemplate    bool         `json:"isTemplate"`
	CreatedByID   string       `json:"createdById"`
	CreatedByName string       `json:"createdByName"`
	UpdatedByID   string       `json:"updatedById"`
	UpdatedByName string       `json:"updatedByName"`
	CreatedAt     time.Time    `json:"createdAt"`
	UpdatedAt     time.Time    `json:"updatedAt"`
}

// MKBArticleListParams holds parameters for listing marketing KB articles
type MKBArticleListParams struct {
	TenantID    string
	ContentType string
	Persona     string
	ContextTag  string
	Status      string
	Query       string
	Limit       int
	HasCursor   bool
	CursorTime  time.Time
	CursorID    string
}

// PitchKitListParams holds parameters for listing pitch kits
type PitchKitListParams struct {
	TenantID      string
	TargetPersona string
	IsTemplate    *bool
	Limit         int
	HasCursor     bool
	CursorTime    time.Time
	CursorID      string
}

// MKBStats holds aggregate statistics for marketing KB articles
type MKBStats struct {
	Total         int            `json:"total"`
	Approved      int            `json:"approved"`
	InReview      int            `json:"inReview"`
	Draft         int            `json:"draft"`
	ByContentType map[string]int `json:"byContentType"`
	ByPersona     map[string]int `json:"byPersona"`
	ByContextTag  map[string]int `json:"byContextTag"`
}

// ValidMKBContentTypes returns all valid content types
func ValidMKBContentTypes() []MKBContentType {
	return []MKBContentType{
		MKBContentTypeMessaging,
		MKBContentTypeCaseStudy,
		MKBContentTypeDeck,
		MKBContentTypeObjection,
		MKBContentTypeROI,
	}
}

// ValidMKBPersonas returns all valid personas
func ValidMKBPersonas() []MKBPersona {
	return []MKBPersona{
		MKBPersonaDirector,
		MKBPersonaPrincipal,
		MKBPersonaTeacher,
		MKBPersonaParent,
		MKBPersonaITAdmin,
		MKBPersonaCountyOfficial,
	}
}

// ValidMKBContextTags returns all valid context tags
func ValidMKBContextTags() []MKBContextTag {
	return []MKBContextTag{
		MKBContextRural,
		MKBContextUrban,
		MKBContextLowConnectivity,
		MKBContextNoISP,
		MKBContextConnected,
		MKBContextPrivate,
		MKBContextPublic,
		MKBContextCBC,
		MKBContextIGCSE,
		MKBContext844,
	}
}

// ValidMKBStatuses returns all valid statuses
func ValidMKBStatuses() []MKBArticleStatus {
	return []MKBArticleStatus{
		MKBStatusDraft,
		MKBStatusReview,
		MKBStatusApproved,
		MKBStatusArchived,
	}
}
